tempesta_fw/sock_clnt.c

```diff
@@ -212,9 +212,30 @@ tfw_sock_clnt_drop(struct sock *sk)
 	return 0;
 }
 
+/*
+ * The hook is executed when a client connection is closed by either
+ * side of the connection.
+ */
+static int
+tfw_sock_clnt_drop(struct sock *sk)
+{
+	return tfw_sock_clnt_do_drop(sk, "connection lost");
+}
+
+/*
+ * The hook is executed when a client connection is terminated due to
+ * an error of any kind.
+ */
+static int
+tfw_sock_clnt_error(struct sock *sk)
+{
+	return tfw_sock_clnt_do_drop(sk, "connection error");
+}
+
 static const SsHooks tfw_sock_clnt_ss_hooks = {
 	.connection_new		= tfw_sock_clnt_new,
 	.connection_drop	= tfw_sock_clnt_drop,
+	.connection_error	= tfw_sock_clnt_error,
```

Comment:
Ingress RST are processed in ```tcp_reset()``` which calls ```sk->sk_error_report()```. Isn't it better to hook the socket callback from SS? I'm confused by the fact that TCP_CLOSE state processed in our ```ss_tcp_state_change()``` can be achieved by many (I think) ways, not always when an error occured. So it seems it'd be more accurate to call ```SsHooks->connection_error``` from ```sk->sk_error_report()``` only.